In [1]:
import pandas as pd
import pingouin as pg

import worldview
from worldview import analysis

# Set display format to 4 decimal places
pd.options.display.float_format = "{:.4f}".format

data_loc = worldview.get_data_dir()
df = pd.read_csv(data_loc / "processed_data.csv")

# One-Way ANOVA - differences in QES by dominant worldview

In [3]:
# Filter out the "multiple_ties" group for this analysis
df_worldview = df[df["dominant_worldview"] != "multiple_ties"]

In [4]:
anova_table, tukey_results = analysis.one_way_anova(
    df_worldview, group_col="dominant_worldview", dv_col="qes_total"
)
print(anova_table)
print(tukey_results)

                            df     sum_sq  mean_sq      F  PR(>F)
C(dominant_worldview)   3.0000  1217.3609 405.7870 8.4006  0.0000
Residual              386.0000 18645.4827  48.3044    NaN     NaN
     Multiple Comparison of Means - Tukey HSD, FWER=0.05      
   group1      group2   meandiff p-adj   lower   upper  reject
--------------------------------------------------------------
integrative      modern   7.4615    0.0  3.5808 11.3422   True
integrative  postmodern    6.409 0.0014  1.9343 10.8837   True
integrative traditional   5.4682 0.0004  1.9458  8.9906   True
     modern  postmodern  -1.0525 0.8741 -4.6446  2.5397  False
     modern traditional  -1.9933 0.1153 -4.2929  0.3064  False
 postmodern traditional  -0.9408 0.8731 -4.1426   2.261  False
--------------------------------------------------------------


There is a statistically significant difference in QES by Dominant Worldview, F(3, 386) = 8.40, p < .001, η2= .06. The tukey tests are used to see where the differences occur (post-hoc tests). The Integrative group scored significantly higher than the Modern group (p < .001), the Postmodern group (p = .001), and the Traditional group (p < .001). There were no statistically significant differences found between the Modern, Postmodern, and Traditional groups (all p values > .11).

Here are the means and SDs that can be used for the writeup: 

In [5]:
df_worldview.groupby("dominant_worldview")["qes_total"].agg(["mean", "std"])

mean    std
dominant_worldview               
integrative        48.8966 8.7560
modern             56.3580 6.6677
postmodern         55.3056 6.8736
traditional        54.3648 6.8158

# One-Way ANOVA - differences in ISI Diffuse Avoidant by Dominant Worldview

In [6]:
anova_table, tukey_results = analysis.one_way_anova(
    df_worldview, group_col="dominant_worldview", dv_col="isi_diffuse_avoidant_total"
)
print(anova_table)

                            df     sum_sq  mean_sq      F  PR(>F)
C(dominant_worldview)   3.0000   139.4605  46.4868 1.1110  0.3445
Residual              386.0000 16151.0139  41.8420    NaN     NaN


No significant difference in ISI diffuse avoidant by dominant worldview (p = 0.344)

# One-Way ANOVA - differences in ISI Informational by Dominant Worldview

In [7]:
anova_table, tukey_results = analysis.one_way_anova(
    df_worldview, group_col="dominant_worldview", dv_col="isi_informational_total"
)
print(anova_table)
print("")
print(tukey_results)

                            df     sum_sq  mean_sq      F  PR(>F)
C(dominant_worldview)   3.0000   359.3092 119.7697 3.1302  0.0257
Residual              386.0000 14769.3806  38.2626    NaN     NaN

     Multiple Comparison of Means - Tukey HSD, FWER=0.05     
   group1      group2   meandiff p-adj   lower  upper  reject
-------------------------------------------------------------
integrative      modern   3.9583 0.0173  0.5044 7.4121   True
integrative  postmodern   3.7021  0.079 -0.2804 7.6846  False
integrative traditional   2.8146  0.096 -0.3204 5.9496  False
     modern  postmodern  -0.2562 0.9969 -3.4532 2.9409  False
     modern traditional  -1.1437  0.474 -3.1904  0.903  False
 postmodern traditional  -0.8875 0.8527 -3.7371 1.9621  False
-------------------------------------------------------------


There is a significant difference in ISI Informational scores by Dominant Worldview, F(3, 386) = 3.13, p = .026, η2 = .02. The Tukey post-hoc tests showed that this is driven by a difference between integrative and modern (p = .017). No other groups have statistically significant differences.  

Here are the means/SDs by groups for writeup:

In [8]:
df_worldview.groupby("dominant_worldview")["isi_informational_total"].agg(
    ["mean", "std"]
)

mean    std
dominant_worldview               
integrative        32.1034 7.1581
modern             36.0617 5.3906
postmodern         35.8056 6.6022
traditional        34.9180 6.2474

# One-Way ANOVA - differences in ISI Normative by Dominant Worldview

In [9]:
anova_table, tukey_results = analysis.one_way_anova(
    df_worldview, group_col="dominant_worldview", dv_col="isi_normative_total"
)
print(anova_table)
print("")
print(tukey_results)

                            df     sum_sq  mean_sq       F  PR(>F)
C(dominant_worldview)   3.0000  2949.6563 983.2188 26.9596  0.0000
Residual              386.0000 14077.4437  36.4701     NaN     NaN

      Multiple Comparison of Means - Tukey HSD, FWER=0.05      
   group1      group2   meandiff p-adj   lower    upper  reject
---------------------------------------------------------------
integrative      modern  -4.4193 0.0044  -7.7913 -1.0473   True
integrative  postmodern   1.2165  0.851  -2.6716  5.1046  False
integrative traditional    -6.82    0.0  -9.8806 -3.7593   True
     modern  postmodern   5.6358    0.0   2.5145  8.7571   True
     modern traditional  -2.4006 0.0111  -4.3988 -0.4024   True
 postmodern traditional  -8.0364    0.0 -10.8185 -5.2544   True
---------------------------------------------------------------


There is a significant difference in ISI Normative scores by Dominant Worldview, F(3, 386) = 26.96, p < .001, η2 = .17. The Tukey post-hoc tests post-hoc test revealed several significant differences between the groups. The Traditional group scored significantly higher than the Integrative group (p < .001), the Modern group (p = .011), and the Postmodern group (p < .001). The Modern group also scored significantly higher than the Integrative group (p = .004) and the Postmodern group (p < .001). No significant difference was found between the Integrative and Postmodern groups (p = .851).

Here are the means/SDs by groups for writeup:

In [10]:
df_worldview.groupby("dominant_worldview")["isi_normative_total"].agg(["mean", "std"])

mean    std
dominant_worldview               
integrative        23.1724 6.5904
modern             18.7531 5.7348
postmodern         24.3889 7.2043
traditional        16.3525 5.8842